### PR TITLE
Split TunnelEntrance and RestrictedArea alerts

### PR DESCRIPTION
**Pull request by korshaknn**
### Description

1. 
 `TunnelEntrance` and `RestrictedArea` alerts coming with route response have line-like geometry and contain `beginCoordinate` + `endCoordinate`.
The same object from `EHorizon` are splitted into 2 objects: `TunnelEntrance` + `TunnelExit`, `RestrictedAreaEntrance` + `RestrictedAreaExit`
It would be useful to refactor it and expose the same objects.
`EHorizon` objects can't be merged to a single one, for example `TunnelEntrance` + `TunnelExit` = `Tunnel` because only one of them may be provided on `EHorizon` update.
So we can split `Directions` alerts into 2 objects.

2.
`Incident`  has `nullable` `IncidentInfo`, it's `nullable` because NN returns all info values in the single object:
```
public final class RoadObjectMetadata implements Serializable {
    ...
    @Nullable
    private final TunnelInfo tunnelInfo;
    @Nullable
    private final BorderCrossingInfo borderCrossingInfo;
    @Nullable
    private final TollCollectionInfo tollCollectionInfo;
    @Nullable
    private final ServiceAreaInfo serviceAreaInfo;
...
```
but, NN says that they guarantee if `type==Incident` ->  `IncidentInfo will be non-null`
the same logic with other types.
so we can use dangerous calls `!!`  on SDK side and expose non-null  values, it looks much better

### Changelog

```
<changelog>Split TunnelEntrance and RestrictedArea alerts, make RoadObjects info values non-null</changelog>
```

### CODE REVIEW

**Review by SiarheiFedartsou — APPROVED** [review]
LGTM

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/NavigatorMapper.kt [hunk 1]
```diff
@@ -654,3 +665,12 @@ internal fun NavigationStatus.prepareSpeedLimit(): SpeedLimit? {
         )
     }
 }
+
+internal sealed class MappedRoadObject {
+    internal data class SingleObject(val roadObject: RoadObject) : MappedRoadObject()
+    internal data class EntranceAndExit(
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/NavigatorMapper.kt:671]: Can these be `private` instead of `internal`?

**Review by Guardiola31337 — APPROVED** [review]
Left one minor comment @korshaknn

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/NavigatorMapper.kt (hunk 1 above)
Comment (korshaknn) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/NavigatorMapper.kt:671]: no, can't be private. 
All mapping functions inside `NavigatorMapper` are `separate` private functions. So, if we make the classes ^^ `private` they will be unavailable for functions.

Comment (Guardiola31337) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/NavigatorMapper.kt:671]: `MappedRoadObject` can be `private` though.

Comment (korshaknn) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/NavigatorMapper.kt:671]: ah, right, I checked with `SingleObject ` and `EntranceAndExit `. but only parent `MappedRoadObject ` can be private

**Pull request merged**